pkg/apis/datavolumecontroller/v1alpha1/types.go

```diff
@@ -42,9 +42,10 @@ type DataVolumeSpec struct {
 
 // DataVolumeSource represents the source for our Data Volume, this can be HTTP, S3 or an existing PVC
 type DataVolumeSource struct {
-	HTTP *DataVolumeSourceHTTP `json:"http,omitempty"`
-	S3   *DataVolumeSourceS3   `json:"s3,omitempty"`
-	PVC  *DataVolumeSourcePVC  `json:"pvc,omitempty"`
+	HTTP   *DataVolumeSourceHTTP   `json:"http,omitempty"`
+	S3     *DataVolumeSourceS3     `json:"s3,omitempty"`
+	PVC    *DataVolumeSourcePVC    `json:"pvc,omitempty"`
+	UPLOAD *DataVolumeSourceUpload `json:"upload,omitempty"`
```

Comment:
Don't you re-run the code generator to accommodate this new type?

Reply:
yes. forgot about that. Thanks!